### ISSUE TITLE
Constraints: Hide unmatched choices

### ISSUE THREAD

**Issue by neumannm** [webanno/webanno#369]
According to a strict reading of the documentation for constraints: "Constraints limit the number of options between which the annotators can choose" - I would like to see the number of options actually reduced to only those choices matched by a given constraint.

**Comment by ChrisBieTUDA** [comment]
if possible, and reasonably easy to do, I vote for making this optional/switchable via a checkbox

**Comment by neumannm** [comment]
Yes this is actually what I had in mind. Because I also see that the current situation has its pros, as is now explained in the new version of the user doc. Should be something similar like the "Hide Un-constraint feature:" checkbox (although I don't know exactly what this is doing).

**Comment by reckart** [comment]
"Hide Un-constraint feature:" means that if a feature is not covered by a constraint rule, then it will be hidden from the UI. I.e. if certain features are only applicable if certain values are set on other features. Example: show "casus" feature only if the annotated word is a noun.

**Comment by neumannm** [comment]
I see, thanks for clarifying.

**Comment by GiantEnemyCrab** [comment]
I still didn't understand "high un-constraint feature"... Would checking that box basically take of what @neumannm requested?

**Comment by sholschneider** [comment]
No. If you check that box, the "allowed" features are shown in bold print while the others are shown in "normal" print. You can still chose features that are not in bold print. Only those features that are not affected by a constraint rule lead to no further possibilities to chose. So, @neumannm 's request is indeed complied when a features is not affected by a constraint rule, but only then.

**Comment by reckart** [comment]
What we have is "Hide Un-constraint feature". 
But what @neumannm wants is "Hide Un-constraint feature *values*".
These are two different things.

**Comment by aakashysharma** [comment]
I am not sure if hiding "non-constraint feature values" was the actual use of constraint while designing it. It was supposed to assist in selection, not eliminate other "feature values".

**Comment by GiantEnemyCrab** [comment]
Right, but I think in this case, there are users like me and @neumannm who rather hide those from users so that it makes the value selection process more efficient by restricting what annotators see. So yes, this would be a welcome option to hide these, although it is a bit different from how it was initially designed.
